src/toolchains.ts

```diff
@@ -0,0 +1,133 @@
+import * as path from 'path';
+import * as core from '@actions/core';
+import * as io from '@actions/io';
+
+import * as fs from 'fs';
+import * as os from 'os';
+
+import { create as xmlCreate } from 'xmlbuilder2';
+import * as constants from './constants';
+import { getBooleanInput } from './util';
+
+export async function configureToolchains(jdkHome: string) {
+  const version = core.getInput(constants.INPUT_JAVA_VERSION);
+  const vendor = core.getInput(constants.INPUT_DISTRIBUTION);
+  const id = core.getInput(constants.INPUT_MVN_TOOLCHAIN_ID) || `${vendor}_${version}`;
+  const settingsDirectory =
+    core.getInput(constants.INPUT_SETTINGS_PATH) || path.join(os.homedir(), constants.M2_DIR);
+  const overwriteSettings = getBooleanInput(constants.INPUT_OVERWRITE_SETTINGS, true);
+
+  await createToolchainsSettings(
+    version,
+    vendor,
+    id,
+    jdkHome,
+    settingsDirectory,
+    overwriteSettings
+  );
+}
+
+export async function createToolchainsSettings(
+  version: string,
+  vendor: string,
+  id: string,
+  jdkHome: string,
+  settingsDirectory: string,
+  overwriteSettings: boolean
+) {
```

Comment:
```suggestion
export async function createToolchainsSettings({
  version: string,
  vendor: string,
  id: string,
  jdkHome: string,
  settingsDirectory: string,
  overwriteSettings: boolean
}) {
```

Since this has a lot of parameters, it would be good to require an object with all the settings.  That way a `string` can't accidentally get passed in the wrong position.

Reply:
Fixed